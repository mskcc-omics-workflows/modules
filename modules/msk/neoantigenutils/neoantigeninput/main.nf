process NEOANTIGENUTILS_NEOANTIGENINPUT {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigen-utils-base:1.1.0':
        'docker.io/mskcc/neoantigen-utils-base:1.1.0' }"

    input:
    tuple val(meta),  path(inputMaf),      path(hlaFile)
    tuple val(meta2), path(phyloWGSsumm),  path(phyloWGSmut),   path(phyloWGSfolder)
    tuple val(meta3), path(mutNetMHCpan),  path(wtNetMHCpan)

    output:
    tuple val(meta), path("*_.json"),                                                  emit: json
    path "versions.yml",                                                               emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def id = task.ext.prefix ?: "${meta.id}"
    def patientid = task.ext.cohort ?: "${meta.id}_patient"
    def cohort = task.ext.cohort ?: "${meta.id}_cohort"

    """
        tree_folder_name=\$(basename -s .zip "${phyloWGSfolder}")
        mkdir \$tree_folder_name
        unzip ${phyloWGSfolder} -d \$tree_folder_name
        gzip -d -c ${phyloWGSsumm} > ${id}.summ.json
        gzip -d -c ${phyloWGSmut} > ${id}.mut.json



        generate_input.py --maf_file ${inputMaf} \
        --summary_file ${id}.summ.json \
        --mutation_file ${id}.mut.json \
        --tree_directory \$tree_folder_name \
        --id ${id} --patient_id ${patientid} \
        --cohort ${cohort} --HLA_genes ${hlaFile} \
        --netMHCpan_MUT_input ${mutNetMHCpan} \
        --netMHCpan_WT_input ${wtNetMHCpan}
        ${args}


        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            neoantigeninput: \$(echo \$(generate_input.py -v))
        END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def id = task.ext.prefix ?: "${meta.id}"
    def patientid =task.ext.cohort ?: "${meta.id}_patient"
    def cohort =task.ext.cohort ?: "${meta.id}_cohort"
    """

        touch ${patientid}_${id}_.json

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            neoantigeninput: \$(echo \$(generate_input.py -v))
        END_VERSIONS
    """
}
