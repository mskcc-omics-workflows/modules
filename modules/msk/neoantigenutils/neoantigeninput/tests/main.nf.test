nextflow_process {

    name "Test Process NEOANTIGENUTILS_NEOANTIGENINPUT"
    script "../main.nf"
    process "NEOANTIGENUTILS_NEOANTIGENINPUT"

    tag "modules"
    tag "modules_nfcore"
    tag "neoantigenutils"
    tag "neoantigenutils/neoantigeninput"

    test("neoantigenutils_neoantigeninput - json,tsv") {

        when {
            params {
                folderPath = 'tst'
            }

            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_maf'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['winners_hla_txt'], checkIfExists: true)
                    ]

                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['test4_summ_json'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['test4_muts_json'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['test4_mutass_zip'], checkIfExists: true)
                    ]

                input[2] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['MUTnetMHCpan_tsv'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['WTnetMHCpan_tsv'], checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }


    test("neoantigenutils_neoantigeninput - json,tsv - stub") {

        options "-stub"

        when {
            params {
                folderPath = 'tst'
            }

            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('temp_test_maf'),
                    file('winners_hla_txt')
                    ]

                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file('test4_summ_json'),
                    file('test4_muts_json'),
                    file(params.folderPath)
                    ]

                input[2] = [
                    [ id:'test', single_end:false ], // meta map
                    file('MUTnetMHCpan.tsv'),
                    file('WTnetMHCpan.tsv')
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
