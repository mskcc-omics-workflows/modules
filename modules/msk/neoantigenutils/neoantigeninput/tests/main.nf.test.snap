{
    "neoantigenutils_neoantigeninput - json,tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4815d54ee3ae0778628a1e51b4b21d9b"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4815d54ee3ae0778628a1e51b4b21d9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-01T16:08:47.586409668"
    },
    "neoantigenutils_neoantigeninput - json,tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,a4442316ba2b6f404b8eb42dd6558eae"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4815d54ee3ae0778628a1e51b4b21d9b"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,a4442316ba2b6f404b8eb42dd6558eae"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4815d54ee3ae0778628a1e51b4b21d9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-01T16:08:31.862365529"
    }
}